### DISCUSSION TITLE
Broadcast from server?

### DISCUSSION THREAD

**Discussion by dstpierre** [jfyne/live#23] [Q&A] [answered]
Hey @jfyne ,

Nice package, I've multiple questions if I may.

1. How would you go about broadcasting from somewhere else in the app, say there's a message queue that process some tasks and would need to broadcast an event to a LiveHandler?
2. On your README you mention that you're starting to use in production, can you share a little your results and experience so far?
3. What's the best way to start contributing?

**Comment by jfyne** [comment, marked as answer]
Hi @dstpierre,

I'm right in the middle of moving apartment so I'm answering on my phone.

1. I've been mulling this around in my mind, thinking about how multiple instances (say on separate pods) would broadcast to each other. Sounds like your external event queue is of a similar class of problem. My immediate thoughts are around broadcast receiver and sender handler funcs. Although I am not clear how this would work in practice, I will need this soon too so I will likely be trying it out. I am very welcome to ideas if you have any.

2. We are replacing an angular frontend, interfacing with a GRPC backend. The amount of boilerplate is vastly reduced,  the complexity of our stack is vastly reduced. We only encountered one big bug in the diffing code which is now fixed. Overall it has been very positive. Also using alpinejs like the PETAL stack, which works great for drop downs menus etc.

3. I'm welcome to any contributions, to me it sounds like solving your problem might be a good place to start. Other things:

-  Tests, I need to write more of them.
-  I'm fairly certain the forms aren't behaving like the Phoenix LiveView forms do.
-  Request animation frame DOM updates.
-  This could be interesting: Rewrite the frontend code in Go and compile it to WASM?
- More complex examples?

Overall I would appreciate more eyes on the code, and discussion on design.

Thanks for getting in touch!

**Reply by dstpierre** [reply]
thanks @jfyne 

It's similar for us, we are starting a decently sized migration (custom jQuery SPA / C# to Go) and were having difficulties accepting a choice for the frontend, Angular came as well as Elm and React. But the thoughts of potentially being able to remove as much JavaScript / TypeScript as possible sounds very attracting.

I remembered having seen this package on reddit couple of months ago, and I'm actually going to create a quick prototype next week to see how it feels.

I'd be happy to contribute if we find anything for our situation, but also generally trying to help as I find Phoenix LiveView very nice, and I'm glad someone created a similar Go implementation.

Regarding broadcasting from multiple nodes, I suppose there will have to be some sort of pub/sub mechanism and each node would broadcast to their sockets. If I recall correctly Phoenix LiveView has a way to listen for topics inside a LiveView component and you may publish from anywhere else into that "topic" and update the model for specifics LiveView that subscribed to that "topic".

I'm planning on having a good look at your implementation this week as well while I prepare the prototype, so I'll probably be able to participate more to discussions and even propose idea once I'm more familiar with the code.

In any case, I'm excited to try it out and will let you know in the following week or so.

Thanks for the good work.

**Reply by jfyne** [reply]
I didn't implement something similar to LiveView PubSub yet because I am keen on going in more of the Go direction in which someone can bring their own preference. I think support for any kind of messaging would be very much desirable, much like at the moment you can bring your own templating, middleware etc.

I look forward to hearing your thoughts in the coming weeks
